File: lab2/Dup.cpp
#include <stdexcept>
#include "FactoryInitializer.h"
#include "Dup.h"

void Dup::Execute(Operands& operands, Tokens& tokens, std::string& output, Reader& reader) {
    int operand1 = operands.Top();
    operands.Push(operand1);
}

namespace {
    FactoryInitializer<Dup> Registration("dup");
}

File: OOP Java/lab1/src/main/java/org/example/commands/Dup.java
package org.example.commands;

import org.example.utility.Operands;
import java.util.StringTokenizer;

public class Dup implements Command{
    @Override
    public void Execute(StringTokenizer tokens, Operands operands) {
        int num1 = operands.Top();
        operands.Push(num1);
    }

    @Override
    public void Check(StringTokenizer tokens, Operands operands) {

    }

}